osu.Game.Rulesets.Catch/UI/Catcher.cs

```diff
@@ -505,13 +505,14 @@ private Vector2 computePositionInStack(Vector2 position, float displayRadius)
             return position;
         }
 
-        private void addLighting(CatchHitObject hitObject, float x, Color4 colour)
+        private void addLighting(PalpableCatchHitObject hitObject, float catchPosition, Color4 colour, JudgementResult result)
         {
-            HitExplosion hitExplosion = hitExplosionPool.Get();
+            PoolableHitExplosion hitExplosion = hitExplosionPool.Get();
             hitExplosion.HitObject = hitObject;
-            hitExplosion.X = x;
-            hitExplosion.Scale = new Vector2(hitObject.Scale);
+            hitExplosion.CatchPosition = catchPosition;
             hitExplosion.ObjectColour = colour;
+            hitExplosion.JudgementResult = result;
+            hitExplosion.CatcherWidth = catchWidth;
```

Comment:
This amount of passing data down is disappointing, but in this case I don't see an easy way of resolving this immediately... (especially with `CatchPosition` - which should arguably be in `JudgementResult` - and `CatcherWidth`).

Reply:
Additionally not having to pass `CatcherWidth` or `CatchPosition` would simplify DI since I wouldn't need to pass two floats, and thus require one of them to be manually named.